indra/newview/app_settings/shaders/class1/gltf/pbrmetallicroughnessV.glsl

```diff
@@ -53,13 +54,37 @@ in vec3 normal;
 in vec4 tangent;
 out vec2 normal_texcoord;
 out vec2 metallic_roughness_texcoord;
+out vec2 occlusion_texcoord;
 out vec3 vary_tangent;
 flat out float vary_sign;
 out vec3 vary_normal;
 vec3 tangent_space_transform(vec4 vertex_tangent, vec3 vertex_normal, vec4[2] khr_gltf_transform, mat4 sl_animation_transform);
 #endif
 
-vec2 texture_transform(vec2 vertex_texcoord, vec4[2] khr_gltf_transform, mat4 sl_animation_transform);
+vec2 gltf_texture_transform(vec2 texcoord, vec4[2] p)
+{
+    texcoord.y = 1.0 - texcoord.y;
+
+    vec2 Scale = p[0].xy;
+    float Rotation = -p[0].z;
+    vec2 Offset = vec2(p[0].w, p[1].x);
+
+    mat3 translation = mat3(1,0,0, 0,1,0, Offset.x, Offset.y, 1);
+    mat3 rotation = mat3(
+        cos(Rotation), sin(Rotation), 0,
+        -sin(Rotation), cos(Rotation), 0,
+        0, 0, 1);
+
+    mat3 scale = mat3(Scale.x,0,0, 0,Scale.y,0, 0,0,1);
+
+    mat3 matrix = translation * rotation * scale;
+
+    vec2 uvTransformed = ( matrix * vec3(texcoord.xy, 1) ).xy;
+
+    uvTransformed.y = 1.0 - uvTransformed.y;
```

Comment:
I'm pretty sure that the offset of 1 is not needed for pure GLTF. I think the offset used in the PBR-on-Blinn-Phong texture transform is meant to accommodate texture animations. Case in point, in textureUtilV.glsl, I have removed the offset for `terrain_texture_transform`

Reply:
if you mean the 1.0-, it is necessary.  There's a corresponding vertical flip in primitive.cpp.  I'm  trying to keep the UVs in CPU memory in GLTF space and not OpenGL space, and I'm trying to keep the glsl shaders as close to stock GLTF as possible.

Reply:
What I mean is, this could work instead:

```
uvTransformed.y = -uvTransformed.y;
```

Reply:
Assuming that's correct, looking at `Primitive::prep` : This implies we will also need to remove the uv offsets of 1 there too.